PR: https://github.com/odoo/odoo/pull/217987

From: 1d30ecfc5681bdf51a60f6447a13415d4142932c
From: Stéphane Diez
Date: 2025-09-04 15:16:28

Breaking data model changes scores: del:3 + add:0, change matches:
-    po_lead = fields.Float(string='Purchase Lead Time', required=True,
-    po_lead = fields.Float(related='company_id.po_lead', readonly=False)
-    use_po_lead = fields.Boolean(

Total Changes: 16

[REF] mrp_*, purchase_*: remove security lead time

We are removing po_lead and manufacturing_lead. Nothing is replacing them but
the user can still use `bom.produce_delay`, `supplierinfo.delay` and
`res_company.days_to_purchase` to apply lead times as needed.

task 4779169

Part-of: odoo/odoo#217987
Related: odoo/enterprise#90534
Related: odoo/upgrade#8192
Signed-off-by: Tiffany Chang (tic) <tic@odoo.com>

================================= pseudo patch: =================================

--- a/addons/purchase/models/res_company.py
+++ b/addons/purchase/models/res_company.py
@@ -7,12 +7,6 @@ from odoo import fields, models
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
-    po_lead = fields.Float(string='Purchase Lead Time', required=True,
-        help="Margin of error for vendor lead times. When the system "
-             "generates Purchase Orders for procuring products, "
-             "they will be scheduled that many days earlier "
-             "to cope with unexpected vendor delays.", default=0.0)
-
     po_lock = fields.Selection([
         ('edit', 'Allow to edit purchase orders'),
         ('lock', 'Confirmed purchase orders are not editable')

--- a/addons/purchase/models/res_config_settings.py
+++ b/addons/purchase/models/res_config_settings.py
@@ -17,20 +17,10 @@ class ResConfigSettings(models.TransientModel):
     module_account_3way_match = fields.Boolean("3-way matching: purchases, receptions and bills")
     module_purchase_requisition = fields.Boolean("Purchase Agreements")
     module_purchase_product_matrix = fields.Boolean("Purchase Grid Entry")
-    po_lead = fields.Float(related='company_id.po_lead', readonly=False)
-    use_po_lead = fields.Boolean(
-        string="Security Lead Time for Purchase",
-        config_parameter='purchase.use_po_lead',
-        help="Margin of error for vendor lead times. When the system generates Purchase Orders for reordering products,they will be scheduled that many days earlier to cope with unexpected vendor delays.")
 
     group_send_reminder = fields.Boolean("Receipt Reminder", implied_group='purchase.group_send_reminder', default=True,
         help="Allow automatically send email to remind your vendor the receipt date")
 
-    @api.onchange('use_po_lead')
-    def _onchange_use_po_lead(self):
-        if not self.use_po_lead:
-            self.po_lead = 0.0
-
     @api.onchange('group_product_variant')
     def _onchange_group_product_variant_purchase(self):
         """If the user disables the product variants -> disable the product configurator as well"""
